app/internal/page/donor/register.go

```diff
@@ -409,3 +426,42 @@ func makeLpaHandle(mux *http.ServeMux, store sesh.Store, defaultOptions handleOp
 		})
 	}
 }
+
+type payHelper struct {
+	logger       Logger
+	sessionStore sessions.Store
+	payClient    PayClient
+	appPublicURL string
+	randomString func(int) string
+}
+
+func (p *payHelper) Pay(appData page.AppData, w http.ResponseWriter, r *http.Request, lpa *page.Lpa) error {
+	createPaymentBody := pay.CreatePaymentBody{
+		Amount:      page.CostOfLpaPence,
+		Reference:   p.randomString(12),
```

Comment:
This hasn't explicitly been said yet but based on the vega call earlier this week I think this will need to be the UID we get back from sending the basic donor details to Sirius to support the separate Sirius processes around matching payments.

Reply:
yeah, well at least now it is only specified in one place when we get round to that